Introduces a new, nest linear layout

jnovinger opened the pull request:
New layout includes separate text views for headline and byline and an
image view for a thumbnail. Uses the Glide library to retrieve images by
URL.

Adds new `LeadArt` data class to aid parsing json for lead art urls.

jnovinger committed:
Add items from DD API feed

- Use okhttp to retrive JSON data
- Use gson to parse JSON into usable objects
- Modify ListActivity to use feed data rather than static list of string

jnovinger committed:
Reimplement list activity with RecyclerView

Also refactor the list item view a bit to include two separate text
views, one for headline and one for byline.

jnovinger committed:
Introduces a new, nest linear layout

New layout includes separate text views for headline and byline and an
image view for a thumbnail. Uses the Glide library to retrieve images by
URL.

Adds new `LeadArt` data class to aid parsing json for lead art urls.

moagrius commented on app/src/main/java/org/novinger/jason/demoreader/datamodels/LeadArt.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * Created by jason on 5/1/16.
+ */
+public class LeadArt {
```
so this is fine, and is probably how i'd do it too, but fwiw you could probably just use a `HashMap` instead of this class

jnovinger replied:
Ah, I was unaware that `gson` would do that. I'll check it out.

moagrius replied:
gson will generally try really hard to fit the model you give it

moagrius commented on app/src/main/res/layout/article_list_item_view.xml:
```diff
@@ -3,25 +3,49 @@
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:textAppearance="?android:attr/textAppearanceMedium"
```
do you understand what's happening here?  also, if you have time, check out "styles"...  i'll give you a quick penny tour...

in `styles.xml` (google where to save it), you can define sets of XML attributes, and inherit from other styles, like so

```
<style name="MyAppTextStyle">
  <item name="android:layout_width">match_parent</style>
  <item name="android:layout_height">wrap_content</style>
</style>
<style name="MyAppBigTextStyle" parent="MyAppTextStyle"><!-- note the "parent", will get all attributes in the parent -->
  <item name="android:textSize">14sp</item>
</style>
```

then you use it for the global `style` attribute, reference it like any other value `@style/NameOfYourStyle`

```
<TextView
  android:id="@+id/bob"
  style="@style/MyAppBigTextStyle" />
```

jnovinger replied:
Ah very cool.

Yeah, I assumed this was a lot like using `<h1>` just to make larger text, rather than using it semantically.

moagrius replied:
i personally almost never use the built in attributes (android:attr) because they're generally system settings, although some times you do need to read them (e.g., action bar height).  i don't think it's bad practice to use them, just normally your designer is going to give you specs for formatting rules like that.  +0/-0

moagrius commented:
+1

jnovinger committed:
Combine condense `getFeedItems()` into `getArticles()`

Unfortunately, I can't seem to make gson serialize a (top-level) JSON
array straight to an `ArrayList`. So, I've moved all the retrieving of
articles logic into one method, `getArticles()`.

We grab the articles as an old-fashion `Array` and immediately add them
to the `ArrayList` we plan to return. In this way, we do our best to
remove any references we have to the original array (partially by having
the stack go away on `return`) allowing as much memory as possible to be
recovered _when_ GC is run.

jnovinger commented:
Closing in order to re-PR against `master`.
